src/Algorithms/CARTTrainer.cpp

```diff
@@ -529,12 +527,10 @@ void CARTTrainer::splitAttributeTables(AttributeTables const& tables, std::size_
 ///Calculates the Gini impurity of a node. The impurity is defined as
 ///1-sum_j p(j|t)^2
 ///i.e the 1 minus the sum of the squared probability of observing class j in node t
-double CARTTrainer::gini(boost::unordered_map<std::size_t, std::size_t>& countMatrix, std::size_t n){
+double CARTTrainer::gini(ClassVector const& countVector, std::size_t n) const {
 	double res = 0;
```

Comment:
res = sum(sqr(countVector/static_cast<double>(n))); //(hopefully, if i did not fuck up in the linear algebra...)

also din't you define a free function gini as well?